packages/identity-sdk/src/viem-sdk.ts

```diff
@@ -0,0 +1,215 @@
+import {
+  Address,
+  PublicClient,
+  WalletClient,
+  parseAbi,
+  SimulateContractParameters,
+  WalletActions,
+} from "viem";
+import { waitForTransactionReceipt } from "viem/actions";
+import { z } from "zod";
+import { compressToEncodedURIComponent } from "lz-string";
+
+import { Envs, FV_IDENTIFIER_MSG2 } from "./constants";
+
+const identityV2ABI = parseAbi([
+  "function addWhitelisted(address account)",
+  "function removeWhitelisted(address account)",
+  "function isWhitelisted(address account) view returns (bool)",
+  "function lastAuthenticated(address account) view returns (uint256)",
+  "function authenticationPeriod() view returns (uint256)",
+]);
+
+const addressSchema = z
+  .string()
+  .regex(/^0x[a-fA-F0-9]{40}$/, "Invalid Ethereum address");
+
+interface IdentityContract {
+  publicClient: PublicClient;
+  contractAddress: Address;
+  abi: any;
+}
+
+interface IdentityExpiryData {
+  lastAuthenticated: bigint;
+  authPeriod: bigint;
+}
+
+export const initializeIdentityContract = (
+  publicClient: PublicClient,
+  contractAddress: Address,
+): IdentityContract => ({
+  publicClient,
+  contractAddress,
+  abi: identityV2ABI,
+});
+
+export const submitAndWait = async (
+  params: SimulateContractParameters,
+  publicClient: PublicClient,
+  walletClient: WalletClient & WalletActions,
+  onHash?: (hash: `0x${string}`) => void,
+) => {
+  try {
+    const [account] = await walletClient.getAddresses();
+    if (!account) throw new Error("No active wallet address found.");
+
+    const { request } = await publicClient.simulateContract({
+      account,
+      ...params,
+    });
+
+    const hash = await walletClient.writeContract(request);
+    onHash?.(hash);
+
+    const receipt = await waitForTransactionReceipt(publicClient, { hash });
+
+    if (!receipt) {
+      throw new Error(`Transaction receipt not found for hash ${hash}`);
+    }
+
+    if (receipt.status !== "success") {
+      throw new Error(`Transaction failed with status ${receipt.status}`);
+    }
+
+    return receipt;
+  } catch (error: any) {
+    console.error("submitAndWait Error:", error);
+    throw new Error(`Failed to submit transaction: ${error.message}`);
+  }
+};
+
+export const isWhitelisted = async (
+  contract: IdentityContract,
+  accountToCheck: Address,
+): Promise<boolean> => {
+  const validation = addressSchema.safeParse(accountToCheck);
+  if (!validation.success) {
+    throw new Error(
+      `isWhitelisted Error: ${validation.error.issues.map((issue) => issue.message).join(", ")}`,
+    );
+  }
+
+  try {
+    const result = await contract.publicClient.readContract({
+      address: contract.contractAddress,
+      abi: contract.abi,
+      functionName: "isWhitelisted",
+      args: [accountToCheck],
+    });
+
+    if (typeof result !== "boolean") {
+      throw new Error(
+        `isWhitelisted Error: Expected boolean, received ${typeof result}`,
+      );
+    }
```

Comment:
why are you verifying the type of blockchain result? also in other functions? simply return the result

Reply:
Its just type safety

Reply:
but why? ok lets say it returns something else? only when it is 'true' then it is relevant any other value will be counted as 'false'